sys/log/include/log/log.h

```diff
@@ -72,11 +86,22 @@ struct log_entry_hdr {
 #define LOG_MODULE_NIMBLE_CTLR      (3)
 #define LOG_MODULE_NIMBLE_HOST      (4)
 #define LOG_MODULE_NFFS             (5)
-#define LOG_MODULE_PERUSER          (64)
+#define LOG_MODULE_MAX              (255)
+
+#define LOG_MODULE_STR(module) \
+    (LOG_MODULE_DEFAULT     == module ? "DEFAULT"     :\
+    (LOG_MODULE_OS          == module ? "OS"          :\
+    (LOG_MODULE_NEWTMGR     == module ? "NEWTMGR"     :\
+    (LOG_MODULE_NIMBLE_CTLR == module ? "NIMBLE_CTLR" :\
+    (LOG_MODULE_NIMBLE_HOST == module ? "NIMBLE_HOST" :\
+    (LOG_MODULE_NFFS        == module ? "NFFS"        :\
+     "UNKNOWN"))))))
```

Comment:
- Macro to string expansion(Stringification) converts the define(constant) into string literal. It does not really give the name of the define. More here: https://gcc.gnu.org/onlinedocs/cpp/Stringification.html

The MACRO is close to the actual definitions of the constants which makes it easier to edit in the future if new modules or levels are added. This way someone doesn't have to go searching for the function to add one more string. C Source file remains intact, only the header file changes.